File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/reset/ClusterResetEvent.java
package com.sequenceiq.cloudbreak.core.flow2.cluster.reset;

import com.sequenceiq.flow.core.FlowEvent;
import com.sequenceiq.flow.event.EventSelectorUtil;
import com.sequenceiq.cloudbreak.reactor.api.event.cluster.StartClusterFailed;
import com.sequenceiq.cloudbreak.reactor.api.event.cluster.StartClusterSuccess;
import com.sequenceiq.cloudbreak.reactor.api.event.resource.ClusterResetResult;

public enum ClusterResetEvent implements FlowEvent {
    CLUSTER_RESET_EVENT("CLUSTER_RESET_TRIGGER_EVENT"),
    CLUSTER_RESET_FINISHED_EVENT(EventSelectorUtil.selector(ClusterResetResult.class)),
    CLUSTER_RESET_FINISHED_FAILURE_EVENT(EventSelectorUtil.failureSelector(ClusterResetResult.class)),

    CLUSTER_RESET_START_CLUSTER_MANAGER_FINISHED_EVENT(EventSelectorUtil.selector(StartClusterSuccess.class)),
    CLUSTER_RESET_START_CLUSTER_MANAGER_FINISHED_FAILURE_EVENT(EventSelectorUtil.failureSelector(StartClusterFailed.class)),

    FINALIZED_EVENT("CLUSTERRESETFINALIZEDEVENT"),
    FAILURE_EVENT("CLUSTERRESETFAILUREEVENT"),
    FAIL_HANDLED_EVENT("CLUSTERRESETFAILHANDLEDEVENT");

    private final String event;

    ClusterResetEvent(String event) {
        this.event = event;
    }

    @Override
    public String event() {
        return event;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/reset/ClusterResetEvent.kt
package com.sequenceiq.cloudbreak.core.flow2.cluster.reset

import com.sequenceiq.cloudbreak.core.flow2.FlowEvent
import com.sequenceiq.cloudbreak.core.flow2.FlowTriggers
import com.sequenceiq.cloudbreak.reactor.api.event.EventSelectorUtil
import com.sequenceiq.cloudbreak.reactor.api.event.cluster.StartAmbariFailed
import com.sequenceiq.cloudbreak.reactor.api.event.cluster.StartAmbariSuccess
import com.sequenceiq.cloudbreak.reactor.api.event.resource.ClusterResetResult

enum class ClusterResetEvent private constructor(private val stringRepresentation: String) : FlowEvent {
    CLUSTER_RESET_EVENT(FlowTriggers.CLUSTER_RESET_TRIGGER_EVENT),
    CLUSTER_RESET_FINISHED_EVENT(EventSelectorUtil.selector(ClusterResetResult::class.java)),
    CLUSTER_RESET_FINISHED_FAILURE_EVENT(EventSelectorUtil.failureSelector(ClusterResetResult::class.java)),

    CLUSTER_RESET_START_AMBARI_FINISHED_EVENT(EventSelectorUtil.selector(StartAmbariSuccess::class.java)),
    CLUSTER_RESET_START_AMBARI_FINISHED_FAILURE_EVENT(EventSelectorUtil.failureSelector(StartAmbariFailed::class.java)),

    FINALIZED_EVENT("CLUSTERRESETFINALIZEDEVENT"),
    FAILURE_EVENT("CLUSTERRESETFAILUREEVENT"),
    FAIL_HANDLED_EVENT("CLUSTERRESETFAILHANDLEDEVENT");

    override fun stringRepresentation(): String {
        return stringRepresentation
    }
}
